server/src/main/java/org/apache/druid/server/http/RulesResource.java

```diff
@@ -181,4 +186,37 @@ private List<AuditEntry> getRuleHistory(
     return auditManager.fetchAuditHistory(AUDIT_HISTORY_TYPE, theInterval);
   }
 
+  /**
+   * Validate rules. Throws an exception if a rule contain an interval that will overshadow another rules' interval.
```

Comment:
I would prefer not using the word overshadow as it has a very specific meaning in Druid. Better use simpler words like "fully cover" or "fully hide" or anything else that works.

Reply:
Heh, sure